### DISCUSSION TITLE
Can't get all User attributes

### DISCUSSION THREAD

**Discussion by migber3200** [DirectoryTree/LdapRecord#446] [General]
Hi.

We have configured active directory and, for each user, a series of data.

To connect to LDAP we are using ldaprecord and we have a problem. It seems that the ldaprecord User object doesn't recognize some of the attributes. When we run hasAttribute on some of them, it always returns false.

If, from the console, we execute the following:
Get-ADUser -Identity [user] -Properties *
returns all the attributes of the LDAP User account. AccountExpirationDate, accountExpires, CN, DistinguishedName, etc...
The problem is that when we have the same user as an ldaprecord User object, it doesn't recognize attributes like extensionAttribute1, extensionAttribute2, extensionAttribute3, SmartcardLogonRequired, etc...

What can we do to make these attributes and their values available from an ldaprecord User object?

Thank you very much.

**Comment by stevebauman** [comment]
Hi @migber3200,

Are those attributes actually set on the user? Or are they empty?

Can you also share your code?

**Reply by migber3200** [reply]
Hi.

They are empty in some cases and set in others.

getLDAPUser searchs and returns the ActiveDirectory\User object
`$LDAPUser = $this->getLDAPUser($model);`
`dd($LDAPUser->hasAttribute('extensionAttribute1')); ` ->Always returns false with some attributes like extensionAttribute1
`$LDAPUser->setAttribute('ExtensionAttribute1', 'cosa');       
$LDAPUser->save();` -> returns error because it does not exists

Thanks.

**Reply by migber3200** [reply]
Any help, please?
